src/test/java/seedu/address/logic/commands/income/AddIncomeCommandTest.java

```diff
@@ -0,0 +1,243 @@
+package seedu.address.logic.commands.income;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIncomes.NUS;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.IncomeList;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.ReadOnlyIncomeList;
+import seedu.address.model.ReadOnlyUserPrefs;
+import seedu.address.model.income.Income;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.IncomeBuilder;
+
+public class AddIncomeCommandTest {
+
+    @Test
+    public void constructor_nullIncome_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddIncomeCommand(null));
+    }
+
+    @Test
+    public void execute_incomeAcceptedByModel_addSuccessful() throws Exception {
+        ModelStubAcceptingIncomeAdded modelStub = new ModelStubAcceptingIncomeAdded();
+        Income validIncome = new IncomeBuilder().build();
+
+        CommandResult commandResult = new AddIncomeCommand(validIncome).execute(modelStub);
+
+        assertEquals(String.format(AddIncomeCommand.MESSAGE_SUCCESS, Messages.formatIncome(validIncome)),
+                commandResult.getFeedbackToUser());
+        assertEquals(Arrays.asList(validIncome), modelStub.incomesAdded);
+    }
+
+    @Test
+    public void execute_duplicateIncome_throwsCommandException() {
+        Income validIncome = new IncomeBuilder().build();
+        AddIncomeCommand addIncomeCommand = new AddIncomeCommand(validIncome);
+        ModelStub modelStub = new ModelStubWithIncome(validIncome);
+
+        assertThrows(CommandException.class,
+                AddIncomeCommand.MESSAGE_DUPLICATE_INCOME, () -> addIncomeCommand.execute(modelStub));
+    }
+
+    @Test
+    public void equals() {
+        Income nus = new IncomeBuilder().withName("Nus").build();
+        Income intern = new IncomeBuilder().withName("Intern").build();
+        AddIncomeCommand addNusCommand = new AddIncomeCommand(nus);
```

Comment:
I recall seeing these as constants, maybe use the constants instead so any changes are captured easily